internal/controllers/core/kubernetesapply/disco.go

```diff
@@ -158,7 +158,7 @@ func (r *Reconciler) toWatchRefs(ka *v1alpha1.KubernetesApply) ([]v1alpha1.Kuber
 		}
 	}
 
-	entities, err := k8s.ParseYAMLFromString(ka.Spec.YAML)
+	entities, err := k8s.ParseYAMLFromString(ka.Status.ResultYAML)
```

Comment:
I've added code to prefer Spec.YAML over Status.ResultYAML. I tried writing a test for that case (i.e., get to this code with Spec.YAML != "" and Status.ResultYAML == ""), but it looks like we already protect against this [further upstream](https://github.com/tilt-dev/tilt/blob/03021311ff2c8c6a409354a3e33367aa39f016be/internal/controllers/core/kubernetesapply/disco.go#L23-L31). Since this is a redundant check (or, ahem, defense in depth), I'm inclined to not worry about the test (the only option that occurred to me is making the fake k8s client configurable to allow apply to return an empty list of entities, but that feels too contrived).